### PR TITLE
Update go-ethereum and ipld-eth-db

### LINKED ISSUE #34: Update to use https://github.com/cerc-io/ipld-eth-db/releases/tag/v5.0.5-alpha schema
E.g. only consider canonical blocks according to this field: https://github.com/cerc-io/ipld-eth-db/blob/9618c0710efb79c37aad8f26e9aaf9b93a9b8b60/db/migrations/00004_create_eth_header_cids_table.sql#L17

### PR DESCRIPTION
**Pull request by roysc**
- for canonicity schema change
- index non-canonical data first in tests
- resolves https://github.com/cerc-io/ipld-eth-db-validator/issues/34

### CODE REVIEW

File: pkg/validator/validator_test.go [hunk 1]
```diff
@@ -119,13 +124,6 @@ func TestStateValidation(t *testing.T) {
 			if err != nil {
 				t.Fatal(err)
 			}
-
-			tx := db.MustBegin()
-			defer tx.Rollback()
-			err = validator.ValidateReferentialIntegrity(tx, i)
```
Comment (i-norden) [inline pkg/validator/validator_test.go:125]: Are we still testing this elsewhere?

**Review by i-norden — COMMENTED** [review]
Looks like this needs a new release of ipld-eth-server to work?

File: pkg/validator/validator_test.go (hunk 1 above)
Comment (roysc) [reply pkg/validator/validator_test.go:125]: Yes, sorry I moved it to the ref integrity tests, but accidentally moved that commit to another branch. Fixed.

**Comment by roysc** [comment]
> Looks like this needs a new release of ipld-eth-server to work?

Doesn't seem to actually. v5.0.0-alpha has calls to `canonical_header_hash` to identify blocks but the function is still defined, so I haven't seen any issues.

**Comment by i-norden** [comment]
> > Looks like this needs a new release of ipld-eth-server to work?
> 
> Doesn't seem to actually. v5.0.0-alpha has calls to `canonical_header_hash` to identify blocks but the function is still defined, so I haven't seen any issues.

Ah I see, we should go ahead and cut a release of ipld-eth-server that includes https://github.com/cerc-io/ipld-eth-server/pull/253 so we aren't dependent on the `canonical_header_hash` function as the intention is to drop that function once it is no longer in use.

**Comment by i-norden** [comment]
Plus there are some other changes in https://github.com/cerc-io/ipld-eth-server/releases/tag/v5.0.1-alpha that are perhaps worth capturing here as well

**Comment by roysc** [comment]
Roger, updated.

**Pull request merged**